test/utils/secret.go

```diff
@@ -0,0 +1,34 @@
+package utils
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+type Secret struct {
+	TestManualConnection bool
+}
+
+func TestSecret(t *testing.T, consumerName string, conf Secret) {
+	t.Run("secrets interface", func(t *testing.T) {
+		if conf.TestManualConnection {
+			SnapConnect(nil,
+				"edgexfoundry:edgex-secretstore-token",
+				"edgex-"+consumerName+":edgex-secretstore-token",
+			)
+		}
+		requireCorrectSecret(t, consumerName)
+	})
+}
+
+func requireCorrectSecret(t *testing.T, consumerName string) {
+	providerSecret, stderr := Exec(t, "sudo cat /var/snap/edgexfoundry/current/secrets/"+consumerName+"/secrets-token.json")
+	require.Empty(t, stderr)
```

Comment:
the Exec will raise an error if it can't ready. Checking stderr shouldn't be necessary.

Reply:
Maybe add a check to make sure providedSecret isn't empty. Otherwise, an empty file copied over will pass the tests.